src/pullrequest.coffee

```diff
@@ -0,0 +1,16 @@
+module.exports = (repo, cb) ->
+  if cb?
+    @get("repos/#{@qualified_repo repo}/pulls", cb)
+  else
+     pr_merge: (prNumber, opts, cb) =>
+      [opts,cb] = [{},opts] unless cb?
+      body =
+        title: opts.title ? "blank title: added by merge-bot automagically"
+        message: opts.message ? "blank message: added by merge-bot automagically"
+        squash: opts.squash ? true
+      @get "repos/#{@qualified_repo repo}/pulls/#{prNumber}", (json) =>
+        body.sha = json.head.merge_commit_sha
+        @put "repos/#{@qualified_repo repo}/pulls/#{prNumber}/merge", body, (data) =>
+          unless data.sha?
+             cb message: data.message
```

Comment:
It looks like the only times the `sha` will be missing are things that would be considered errors, so in this case it probably makes more sense to invoke the error handler instead of a normal response. I *think* you can do it like so:

```coffeescript
@_errorHandler error: data.message
```